opengrok-tools/src/test/python/test_mirror.py

```diff
@@ -109,3 +114,59 @@ def test_valid_config():
     assert check_configuration({PROJECTS_PROPERTY:
                                 {"foo": {PROXY_PROPERTY: True}},
                                 PROXY_PROPERTY: "proxy"})
+
+
+@pytest.mark.skipif(not os.name.startswith("posix"), reason="requires posix")
+def test_incoming_retval(monkeypatch):
+    """
+    Test that the special CONTINUE_EXITVAL value bubbles all the way up to
+    the mirror.py return value.
+    """
+
+    class MockResponse:
+
+        # mock json() method always returns a specific testing dictionary
+        @staticmethod
+        def json():
+            return {"indexed": "true"}
+
+        @staticmethod
+        def raise_for_status():
+            pass
+
+    with tempfile.TemporaryDirectory() as source_root:
+        repo_name = "parent_repo"
+        repo_path = os.path.join(source_root, repo_name)
+        cloned_repo_name = "cloned_repo"
+        cloned_repo_path = os.path.join(source_root, cloned_repo_name)
+        project_name = "foo"    # does not matter for this test
+
+        os.mkdir(repo_path)
+
+        def mock_get_repos(*args, **kwargs):
+            from opengrok_tools.scm.repofactory import get_repository
```

Comment:
any special reason why the import is here?

Reply:
It is used just for this one function.

Reply:
however this function is executed always when this file is executed so it does not bring anything extra

Reply:
more it hides the actual runtime dependency for each run

Reply:
okay, will group them in the beginning